### PR TITLE
fix(ChatDisplay/index.tsx): users are only able to download a protocol

### PR DESCRIPTION
**Pull request by Elyorcv**
# Overview
Previously, when users clicked the download button, it created a Python file even for an empty string.
Now, the system downloads only valid protocols that are not single-line and must include `def run(`.

Closes AUTH-1064

## Test Plan and Hands on Testing
- Go to OpentronsAI.
- Generate a protocol and download it. You should see a Python file containing the protocol.
- Simulate the protocol and get feedback.
- Click Download again, but this time, you should not be able to download the file if the protocol is invalid.

## Changelog
- Modified the functionality of the Download button to ensure only valid protocols can be downloaded.

## Review requests
Look at the section `Test Plan and Hands on Testing`

## Risk assessment
Low

### CODE REVIEW

File: opentrons-ai-client/src/molecules/ChatDisplay/__tests__/ChatDisplay.test.tsx [hunk 1]
```diff
@@ -110,6 +116,24 @@ describe('ChatDisplay', () => {
     })
   })
 
+  it('should not call trackEvent when download button is clicked', () => {
```
Comment (y3rsh) [inline opentrons-ai-client/src/molecules/ChatDisplay/__tests__/ChatDisplay.test.tsx:119]: Thank you for adding a test!

File: opentrons-ai-client/src/molecules/ChatDisplay/index.tsx [hunk 2]
```diff
@@ -106,7 +106,18 @@ export function ChatDisplay({ chat, chatId }: ChatDisplayProps): JSX.Element {
 
   const handleFileDownload = (): void => {
     const lastCodeBlock = document.querySelector(`#${chatId}`)
-    const code = lastCodeBlock?.textContent ?? ''
+    const code = lastCodeBlock?.textContent?.trim() ?? ''
```
Comment (y3rsh) [inline opentrons-ai-client/src/molecules/ChatDisplay/index.tsx:109]: The logic here makes sense and I am guessing that we instruct the LLM to place the protocol contents in the last code block of the response?  Feels a bit brittle but we do not need to address in this PR.

**Review by y3rsh — APPROVED** [review]
1 Question but not a blocker.

File: opentrons-ai-client/src/molecules/ChatDisplay/index.tsx (hunk 2 above)
Comment (Elyorcv) [reply opentrons-ai-client/src/molecules/ChatDisplay/index.tsx:109]: For this version, we assume that there is only one code block.

**Pull request merged**